fendermint/vm/interpreter/src/fvm/state/exec.rs

```diff
@@ -235,6 +235,23 @@ where
         self.execute_message(msg, ApplyKind::Implicit)
     }
 
+    pub fn execute_read_only(&mut self, msg: Message) -> ExecResult {
+        if let Err(e) = msg.check() {
+            return Ok(check_error(e));
+        }
+
+        let raw_length = fvm_ipld_encoding::to_vec(&msg).map(|bz| bz.len())?;
```

Comment:
We use this in 2 different place now. Maybe you can extract to `compute_raw_length` func and use that instead?